### PR TITLE
added provider flag for config generation

### PR DESCRIPTION
**Pull request by DStorck**
defaults to aws

### CODE REVIEW

**Comment by DStorck** [comment]
Don't merge yet but feel free to comment - I am going up to update the documentation to reflect this flag

**Comment by DStorck** [comment]
I am currently working on another branch where several internal parts of generateCmd have been extracted to make testing easier. This new branch will include the same features as this PR, but it will look a little different and will include a test.

**Comment by DStorck** [comment]
Okay, ready to review.  Test was added to check that there is a default provider of 'aws' if provider flag not passed, that providerConfig and generateConfig variables are set in preRun, and that the specified provider will grab the correct default config file.  I was thinking of changing config.yaml to aws-config.yaml, does anyone have any opinions on that?

**Comment by leahnp** [comment]
I vote yes on changing the config to aws-config.

**Comment by DStorck** [comment]
@leahnp The file created is still config.yaml no matter what since k2cli is looking for that file, but I was thinking that down the road as we have more providers and more default configs, it would be good to have aws-config, bare-metal.config, gke-config, joyent-config in the filesystem to make it more obvious. Not really an issue though until we get more provider support.

**Comment by leahnp** [comment]
@DStorck agreed!

File: cmd/generate.go [hunk 1]
```diff
@@ -16,35 +16,47 @@ package cmd
...
 	SilenceUsage: true,
 	Long:         `Generate a K2 configuration file at the specified location`,
-	PreRunE: func(cmd *cobra.Command, args []string) error {
-		if len(args) > 0 {
-			generatePath = os.ExpandEnv(args[0])
-		} else {
-			generatePath = os.ExpandEnv("$HOME/.kraken/config.yaml")
-		}
+  PreRunE: preRunEFunc,
+  Run: runFunc,
```
Comment (coffeepac) [inline cmd/generate.go]: go has a strongly encouraged whitespace formatting rule.  Can you please run 'go fmt' on all files in this project and see what happens?  I would expect a lot of whitespace changes.

File: cmd/generate_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		args[0] = "$HOME/sandbox/fake-config.tmp"
+	  preRunEFunc(nil, args)
+
+		if configPath == "" || generatePath == "" {
+			t.Error("Expected configPath and generatePath to be set")
+		}
+}
+
+func TestAWSProviderCreatesAWSconfig(t *testing.T) {
+	if configPath != "ansible/roles/kraken.config/files/config.yaml " {
```
Comment (coffeepac) [inline cmd/generate_test.go]: this function should be caling preRunEFunc(nil, nil) to ensure that calling that function without explicitly setting provider operates as expected.

**Pull request merged**